kolibri/plugins/coach/assets/src/views/channel-list-page/index.vue

```diff
@@ -49,8 +49,13 @@
     },
     methods: {
       reportLink(channelId) {
+        const linkTargets = {
+          [PageNames.RECENT_CHANNELS]: PageNames.RECENT_ITEMS_FOR_CHANNEL,
+          [PageNames.TOPIC_CHANNELS]: PageNames.TOPIC_CHANNEL_ROOT,
+          [PageNames.LEARNER_CHANNELS]: PageNames.LEARNER_CHANNEL_ROOT,
+        };
         return {
-          name: PageNames.RECENT_ITEMS_FOR_CHANNEL,
+          name: linkTargets[this.pageName],
           params: {
```

Comment:
check out the latest coach app.js for the specific parameters that each page takes.

Channel root pages only take channel and class IDs